### PR TITLE
Bringing back Full Framework version of the task library to be able to build in razzle and linux

### PR DESCRIPTION
**Pull request by joperezr**
Bringing back Full Framework version of the task library to be able to build in razzle and linux.

When we moved this Task library to .Net Core we verified that it would be able to be loaded also when using full framework. What we didn't validate, is that it requires full framework 4.6 which means that it won't work in linux(mono msbuild doesn't support 4.6) nor Razzle builds(razzle runs with full framework 4.5). Because of this, after upgrading the buildtools version broke corefx repo in both environments. This change will add back the 4.5 version of the task library and store it under the 'net45' folder inside the package, so that razzle and linux builds can load that instead of the .Net Core version.

### CODE REVIEW

**Comment by joperezr** [comment]
cc @weshaggard

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <UsingTask TaskName="SendToEventHub" AssemblyFile="$(ToolsDir)net45\Microsoft.DotNet.Build.CloudTestTasks.dll"/>
   <UsingTask TaskName="UploadToAzure" AssemblyFile="$(ToolsDir)net45\Microsoft.DotNet.Build.CloudTestTasks.dll"/>
   <UsingTask TaskName="WriteItemsToJson" AssemblyFile="$(ToolsDir)net45\Microsoft.DotNet.Build.CloudTestTasks.dll"/>
-  <UsingTask TaskName="ZipFileCreateFromDirectory" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="ZipFileCreateFromDirectory" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: In one of the targets files in BuildTools you should default BuildToolsTaskDir to ToolsDir so that not everyone using build tools is forced to set this independently in their repo.

File: src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    <SchemaVersion>2.0</SchemaVersion>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Microsoft.DotNet.Build.Tasks</RootNamespace>
+    <AssemblyName>Microsoft.DotNet.Build.Tasks</AssemblyName>
+    <TargetFrameworkIdentifier>.NETFramework</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <ResolveNuGetPackages>false</ResolveNuGetPackages>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: If possible I would still use the project.json even for the desktop version as opposed to the old packages.config.

File: src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    <SkipSigning>true</SkipSigning>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="..\Microsoft.DotNet.Build.Tasks\Delegates.cs">
+	
+	</Compile>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: Any ideas why there is an end tag and why it is formatted like this? You should be able to simply close it off with "/>".

File: src/nuget/Microsoft.DotNet.BuildTools.nuspec [hunk 4]
```diff
@@ -19,7 +19,16 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="Microsoft.DotNet.Build.CloudTestTasks\*.dll" target="lib\net45" />
+    <file src="Microsoft.DotNet.Build.Tasks.net45\*.dll" target="lib\net45" />
+    <file src="Microsoft.DotNet.Build.CloudTestTasks\Microsoft.Azure.KeyVault.Core.dll" target="lib\net45" />
```
Comment (weshaggard) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: Why did you need to expand this instead of using *.dll now? Is it because there are file conflicts?

File: src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj (hunk 3 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: No idea why it got imported like this, good catch, I'll fix it

File: src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets (hunk 1 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets]: Fixed.

File: src/nuget/Microsoft.DotNet.BuildTools.nuspec (hunk 4 above)
Comment (joperezr) [inline src/nuget/Microsoft.DotNet.BuildTools.nuspec]: There are three dependencies that exist in both the cloudTestTasks and the net45 task library, they are Newtonsoft.Json, System.Reflection.Metadata and System.Collections.Immutable. This causes no issues for now as they both projects have the same version of all three assemblies. When this stops being true, we can start dividing each project into a different folder within net45.

**Comment by ericstj** [comment]
Can you make a more verbose commit message to explain the problem?

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/toolruntime.targets [hunk 5]
```diff
@@ -1,7 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <UsingTask TaskName="PrereleaseResolveNuGetPackageAssets" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <PropertyGroup>
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/toolruntime.targets]: I think this needs to be in a more core location like the Build.Common.targets itself.

File: src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj [hunk 6]
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: Lets avoid this duplication.  Instead can we import the other project so that we don't duplicate?

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/toolruntime.targets (hunk 5 above)
Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/toolruntime.targets]: Will move.

File: src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj (hunk 6 above)
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: If you want you can check in with the duplication and I will come back and clean it up.

Comment (joperezr) [inline src/Microsoft.DotNet.Build.Tasks.net45/Microsoft.DotNet.Build.Tasks.net45.csproj]: I merged this PR to unblock stakeholders and I will send out a new PR shortly with this refactoring. Thanks Eric!
